### ISSUE TITLE
output graph @base 

### ISSUE THREAD

**Issue by fils** [TopQuadrant/shacl#45]
I'm very new to SHACL but interested in exploring its use to validate JSON-LD data graphs.  
I'm able to convert the JSON-LD to TTL and process them via a shape graph using your code (wrapping them in a simple RESTful service).

However, the results of the SHACL process look like the example at the bottom of this issue.  

I generate the shape and data models with a line like

```java
 shapesModel.read(fis2, "urn:dummy", org.apache.jena.util.FileUtils.langTurtle);
```
and another for the dataModel.  

and process with 
```java
 Resource report = ValidationUtil.validateModel(dataModel, shapesModel, false);
```

The result has a @base of example.org/random as seen below.   If I follow this correctly I'm essentially getting a blank node as a subject for these triples.  (Note..  I'm likely NOT following this correctly).  

Should I be able to specify or use the @base to arrive at a non blank subject?  Or is it part of the nature of SHACL that the results have a blank node subject?  What is the role of "urn:dummy" in the Jena memory model?

I'm trying to process around 50K JSON-LD data graphs (schema.org type Dataset)  and generate a graph of their processing against shapes to generate an assessment back to the providers.    So being able to associate the results back to the input data graph would be nice.  

Obviously the "focusNode" provides that to a degree, I understand.  Perhaps that is the manner in which to do this?   My only concern is that some of the providers don't provide an explicit @id in thier JSON-LD which might not work in that location then.  (not sure.  need to test this)

Any guidance or suggested reading/reference is welcome.  

Thanks!

```
@base          <http://example.org/random> .
[ a       <http://www.w3.org/ns/shacl#ValidationReport> ;
  <http://www.w3.org/ns/shacl#conforms>
          false ;
  <http://www.w3.org/ns/shacl#result>
          [ a       <http://www.w3.org/ns/shacl#ValidationResult> ;
            <http://www.w3.org/ns/shacl#focusNode>
                    <http://opencoredata.org/id/dataset/bcd15975-680c-47db-a062-ac0bb6e66816> ;
            <http://www.w3.org/ns/shacl#resultMessage>
                    "Less than 1^^http://www.w3.org/2001/XMLSchema#integer values" ;
            <http://www.w3.org/ns/shacl#resultPath>
                    <http://schema.org/citation> ;
            <http://www.w3.org/ns/shacl#resultSeverity>
                    <http://www.w3.org/ns/shacl#Violation> ;
            <http://www.w3.org/ns/shacl#sourceConstraintComponent>
                    <http://www.w3.org/ns/shacl#MinCountConstraintComponent> ;
            <http://www.w3.org/ns/shacl#sourceShape>
                    []
          ]
] .
```

**Comment by HolgerKnublauch** [comment]
I think @base is a red herring on what you are trying to do (it's just a Turtle serialization convention). The blank node for each sh:ValidationRecord comes from the constructor of ValidationEngine, which sets this.report to be reportModel.createResource(...) unless it is passed in. So you could call this constructor via the ValidationEngineFactory with an already given URI resource if you want to assign it a URI. You could also attach other metadata to the generated sh:ValidationReport resource, e.g. to track what graph or JSON-LD file it was about. But you are also right that the focusNode would sometimes provide enough tracking info. I am not sure how JSON-LD handles the ID generation of unnamed JSON nodes, so I guess that would be a question to JSON-LD expert, on how to ensure that URIs are produced in a consistent manner.

(Hint: before printing the result Model, set prefix "sh" to the SHACL namespace to make the report more readable).

**Comment by fils** [comment]
@HolgerKnublauch Thanks!   this helped a lot!   I put my Java code I am working on at https://gist.github.com/fils/c5b20773f0a4e73a2e807412f492df75   I haven't done any Java coding for years and I wasn't that good even then..  so this likely is kinda sad looking.  

Based on your comment above I was able to resolve how to put in some new triples.   For now I just used SKOS:related and then dropped in a URI value for that.   I'll pipeline that data later while I evaluate the triples.

Converting them to quads of course we are still all blank nodes at the top level.  For this case not really an issue but I will have to pipe them into a triple store to work with.     My main goal is to be able to resolve failures into a report for facilities so that an see where their data graphs are not meeting the patterns defined in the shape graphs.   So I think we are really close here.  

Any further comments or observations welcome of course.

Thanks again..  would never have found the right path without your comments.  

Doug
